src/libopenrave/collisionchecker.cpp

```diff
@@ -78,105 +78,84 @@ void CollisionPairInfo::SwapFirstSecond()
     }
 }
 
+static void _SetCollisionBodyLinkGeomName(std::string& bodyLinkGeomName, const std::string& bodyname, const std::string& linkname, const std::string& geomname)
+{
+    bodyLinkGeomName = bodyname;
+    bodyLinkGeomName.push_back(' ');
+    bodyLinkGeomName += linkname;
+    bodyLinkGeomName.push_back(' ');
+    bodyLinkGeomName += geomname;
+}
+
 void CollisionPairInfo::SetFirstCollision(const std::string& bodyname, const std::string& linkname, const std::string& geomname)
 {
-    bodyLinkGeom1Name = bodyname;
-    bodyLinkGeom1Name.push_back(' ');
-    bodyLinkGeom1Name += linkname;
-    bodyLinkGeom1Name.push_back(' ');
-    bodyLinkGeom1Name += geomname;
+    _SetCollisionBodyLinkGeomName(bodyLinkGeom1Name, bodyname, linkname, geomname);
 }
 
 void CollisionPairInfo::SetSecondCollision(const std::string& bodyname, const std::string& linkname, const std::string& geomname)
 {
-    bodyLinkGeom2Name = bodyname;
-    bodyLinkGeom2Name.push_back(' ');
-    bodyLinkGeom2Name += linkname;
-    bodyLinkGeom2Name.push_back(' ');
-    bodyLinkGeom2Name += geomname;
+    _SetCollisionBodyLinkGeomName(bodyLinkGeom2Name, bodyname, linkname, geomname);
 }
 
-void CollisionPairInfo::ExtractFirstBodyName(string_view& bodyname) const
+inline void _ExtractBodyName(const std::string& bodyLinkGeomName, string_view& bodyname)
```

Comment:
I'm assuming this `inline` is meant as an optimizer hint - for modern compilers it does not have this effect; `inline` semantics are to allow multiple definitions of the same function. If you _really_ want to force code inlining you can use `__attribute__((always_inline))` but generally this is best left to the optimizer.

Reply:
Thanks for your comment!

The functions in this file have originally `inline`.
So, I also do the same thing for the newly added function.

I'm ok to let the optimizer decide it.